pkg/cloudprovider/cloudprovider.go

```diff
@@ -385,7 +385,10 @@ func (c *CloudProvider) instanceToNodeClaim(i *instance.Instance, instanceType *
 
 	if instanceType != nil {
 		for key, req := range instanceType.Requirements {
-			if req.Len() == 1 {
+			// Only add the label if there is a single possible value. Counter examples include zone and reservation id. We need
+			// an explicit check for reservation id because even if there is a single value, if we launched into spot or OD it
+			// won't be used.
+			if req.Len() == 1 && req.Key != cloudprovider.ReservationIDLabel {
```

Comment:
It's not entirely clear to me why this check is here now -- can you help me understand? The comment doesn't make it crystal clear right now IMO

Reply:
Discussed offline: I think this is clear to me but we agreed that we need a generalized approach to solve the offering/instance type requirements problem that this highlights.

We agreed to take a fix to update this comment